tests/test_dictionary.py

```diff
@@ -18,15 +18,36 @@ def setup(self):
       external_search.insert(doc)
 
   def test_definition(self):
-    # Get the example search result for word 'dictionary'.
-    (ignore, ignore2, ignore3, doc) = self.dictionary.external_search.elastic_search_results()
-    definitions = doc["senses"]
-    word = 'dictionary'
-
-    eq_(
-      self.dictionary.definition(word),
-      dict(word=word, definitions=definitions)
-    )
+    url = "/dictionary/definition/English"
+    with app.test_request_context(url) as c:
+      # Get the example search result for word 'dictionary'.
+      (ignore, ignore2, ignore3, doc) = self.dictionary.external_search.elastic_search_results()
+      definitions = doc["senses"]
+      for definition in definitions:
+        definition["pos"] = doc["pos"]
+        definition["metadata"] = definition["glosses"]
+        del definition["glosses"]
+      word = 'dictionary'
+
+      definition_json = self.dictionary.definition(word)
+      # Don't worry about testing the time...
+      del definition_json["metadata"]["modified"]
+
+      eq_(
+        definition_json["metadata"],
+        {
+          'title': ('Definitions for dictionary',),
+          '@type': 'http://schema.org/DefinedTerm',
+          'language': 'English',
+          'name': 'dictionary'
+        }
+      )
+      # eq_(definition_json['links'],
+      #   [{'rel': 'self', 'href': url, 'type': 'application/opds+json'}]
+      # )
+      # eq_(definition_json["definitions"],
+      #   definitions
+      # )
```

Comment:
I'd expect this to work; can you uncomment it?

Reply:
Yeah, for the `links` test I am missing a trailing slash at the end of the `url` variable.
But for testing the definitions, it doesn't like that properties are out of order:
```
[{'metadata': 'To look up in a dictionary.', 'tags': ['transitive'], 'pos': 'verb'}, ...] != [{'tags': ['transitive'], 'pos': 'verb', 'metadata': ['To look up in a dictionary.']}, ...]
```

Reply:
I think the difference here is `metadata` being a string rather than a list.

But per my other comment this will change anyway when `metadata` becomes a dictionary.

Reply:
Oh, missed that. Thanks. But also, I don't know why the title which is a string gets converted to a tuple on line 39 in this same file.